pkg/operator/ceph/file/filesystem.go

```diff
@@ -189,33 +189,58 @@ func (f *Filesystem) doFilesystemCreate(context *clusterd.Context, cephVersion c
 		return fmt.Errorf("Cannot create multiple filesystems. Enable %s env variable to create more than one", client.MultiFsEnv)
 	}
 
-	logger.Infof("Creating filesystem %s", f.Name)
-	err = client.CreatePoolWithProfile(context, clusterName, *f.metadataPool, appName)
+	poolNames, err := client.GetPoolNamesByID(context, clusterName)
 	if err != nil {
-		return fmt.Errorf("failed to create metadata pool '%s': %+v", f.metadataPool.Name, err)
+		return fmt.Errorf("failed to get pool names. %+v", err)
+	}
+
+	logger.Infof("Creating filesystem %s", f.Name)
+
+	// Make easy to locate a pool by name and avoid to repeat searchs
+	reversed_pool_map := make(map[string]int)
+	for key, value := range poolNames {
+		reversed_pool_map[value] = key
+	}
+
+	pools_created := false
+	if _, pool_found := reversed_pool_map[f.metadataPool.Name]; !pool_found {
+		pools_created = true
+		err = client.CreatePoolWithProfile(context, clusterName, *f.metadataPool, appName)
+		if err != nil {
+			return fmt.Errorf("failed to create metadata pool '%s': %+v", f.metadataPool.Name, err)
+		}
 	}
 
 	var dataPoolNames []string
 	for _, pool := range f.dataPools {
 		dataPoolNames = append(dataPoolNames, pool.Name)
-		err = client.CreatePoolWithProfile(context, clusterName, *pool, appName)
-		if err != nil {
-			return fmt.Errorf("failed to create data pool %s: %+v", pool.Name, err)
-		}
-		if pool.Type == model.ErasureCoded {
-			// An erasure coded data pool used for a filesystem must allow overwrites
-			if err := client.SetPoolProperty(context, clusterName, pool.Name, "allow_ec_overwrites", "true"); err != nil {
-				logger.Warningf("failed to set ec pool property: %+v", err)
+		if _, pool_found := reversed_pool_map[pool.Name]; !pool_found {
+			pools_created = true
+			err = client.CreatePoolWithProfile(context, clusterName, *pool, appName)
+			if err != nil {
+				return fmt.Errorf("failed to create data pool %s: %+v", pool.Name, err)
+			}
+			if pool.Type == model.ErasureCoded {
+				// An erasure coded data pool used for a filesystem must allow overwrites
+				if err := client.SetPoolProperty(context, clusterName, pool.Name, "allow_ec_overwrites", "true"); err != nil {
+					logger.Warningf("failed to set ec pool property: %+v", err)
+				}
 			}
 		}
 	}
 
-	// create the filesystem
-	if err := client.CreateFilesystem(context, clusterName, f.Name, f.metadataPool.Name, dataPoolNames); err != nil {
+	// create the filesystem ('fs new' needs to be forced in order to reuse pre-existant pools)
+	// if only one pool is created new it wont work (to avoid inconsistencies)
+	force := !pools_created
+	if force {
+		logger.Infof("Filesystem %s will reuse pre-existant pools", f.Name)
```

Comment:
nit: `pre-existing` instead of `pre-existant`?

Reply:
how about logging this inside `client.CreateFilesystem()` instead of here?

Reply:
Fixed and moved the log comment. thx!